### PR TITLE
Mission destination ding

### PR DESCRIPTION
**Pull request by a-random-lemurian**
**Feature**

This PR addresses the bug/feature described in issue #10698 

## Summary
Adds an option to play a ding on entering a system where you have a mission, in addition to the green text you already get.

## Testing Done
Did a few missions and listened for the ding and made sure the settings were correct.

## Performance Impact
Negligible.

## Extra notes
- Probably might not want to use the default fail sound for this. It is only a placeholder.
- Not sure if the current wording of the setting ("Play ding on destination") is the best.

### CODE REVIEW

File: source/AI.cpp [hunk 1]
```diff
@@ -4171,6 +4171,9 @@ void AI::MovePlayer(Ship &ship, Command &activeCommands)
 			}
 			message += " in the system you are jumping to.";
 			Messages::Add(message, Messages::Importance::Info);
+
+			if(Preferences::Has("Play ding on destination"))
+				Audio::Play(Audio::Get("fail"));
```
Comment (TomGoodIdea) [inline source/AI.cpp:4176]: I'm not sure if the failure sound fits here.

File: source/Preferences.cpp [hunk 2]
```diff
@@ -172,6 +172,7 @@ void Preferences::Load()
 	settings["Ship outlines in HUD"] = true;
 	settings["Extra fleet status messages"] = true;
 	settings["Target asteroid based on"] = true;
+	settings["Play ding on destination"] = false;
```
Comment (TomGoodIdea) [inline source/Preferences.cpp:175]: no need to initialize default-to-off preferences
```suggestion
```

File: source/AI.cpp (hunk 1 above)
Comment (tibetiroka) [reply source/AI.cpp:4176]: I think it works pretty well in the game.

Comment (a-random-lemurian) [reply source/AI.cpp:4176]: > I think it works pretty well in the game.

The sound plays if there's something important the player has to care about, and well, having to land on a planet for a mission is pretty important, so the only thing we might need is to change the name from "fail" to something more general in another PR.

File: data/_ui/tooltips.txt [hunk 3]
```diff
@@ -1485,3 +1485,6 @@ tip "Date format"
 
 tip "Show parenthesis"
 	`Toggle parenthesis around the trade prices in the ports map mode.`
+
+tip "Play ding on destination"
```
Comment (tibetiroka) [inline data/_ui/tooltips.txt:1489]:
```suggestion
tip "Notify on destination"
```

File: data/_ui/tooltips.txt [hunk 4]
```diff
@@ -1485,3 +1485,6 @@ tip "Date format"
...
+	`If you enter a system with a planet you must land on for a mission, plays a ding.`
```
Comment (tibetiroka) [inline data/_ui/tooltips.txt:1490]:
```suggestion
	`If you enter a system with a planet you must land on for a mission, plays a notification sound.`
```

**Review by xobes — COMMENTED** [review]
Could / should this be on by default?  Probably after a new sound is introduced.

This one says it's public domain: https://opengameart.org/content/beep-tone-sound-sfx
Same here: https://opengameart.org/content/ui-sounds-0 -- I like the 2nd and third sounds.

**Review by TomGoodIdea — APPROVED** [review]
code side ok

**Review by TheGiraffe3 — APPROVED** [review]
I think `fail` works alright.

**Comment by a-random-lemurian** [comment]
(Not sure how to request a review from @Koranir - GitHub won't let me for some unknown reason)

**Comment by TomGoodIdea** [comment]
You can only request reviews from people that have reviewed your PR previously, unless you have triage or higher permissions. ~~I would do it for you, but I'm on mobile right now, and of course it doesn't have this feature~~

File: source/Preferences.cpp [hunk 5]
```diff
@@ -222,6 +225,8 @@ void Preferences::Load()
 			previousSaveCount = max<int>(3, node.Value(1));
 		else if(node.Token(0) == "alt-mouse turning")
 			settings["Control ship with mouse"] = (node.Size() == 1 || node.Value(1));
+		else if(node.Token(0) == "notification settings")
+			dateFormatIndex = max<int>(0, min<int>(node.Value(1), NOTIF_OPTIONS.size() - 1));
```
Comment (tibetiroka) [inline source/Preferences.cpp:229]: Should this really be dateFormatIndex for "notification settings"?

Comment (a-random-lemurian) [reply source/Preferences.cpp:229]: sloppy copy-paste. gonna fix it

**Review by tibetiroka — COMMENTED** [review]
I think you forgot to save the preference.

**Comment by a-random-lemurian** [comment]
> I think you forgot to save the preference.

and i now have it saved

**Review by tibetiroka — APPROVED** [review]

**Pull request merged**